### PR TITLE
Adds clear functionality

### PR DESCRIPTION
**Pull request by gomorizsolt**
Clear functionality now works, when clicking on button the checklist is a clean array, removing URL parameters and updating the "checked" props. I've used **componentWillUpdate** lifecycle hook, because the constructor runs only once and not updating the checked field inside the state - without this we can't see the changes, need to refresh the page and that's not the best way.  LocalStorage is also cleared, of course.

But I'm interested in any other suggestions, maybe it costs a lot to run this lifecycle hook.

Waiting for button style suggestions.

### CODE REVIEW

**Comment by Kexort** [comment]
Preview available here: https://s3.eu-central-1.amazonaws.com/kexort/issue-38_clear_functionality/index.html

**Comment by thisismydesign** [comment]
Let's use Material UI trash icon: https://material-ui.com/demos/buttons/#icon-buttons

File: src/list.js [hunk 1]
```diff
@@ -48,17 +48,25 @@ class List extends React.Component {
...
     let checklist = this.state.checklist;
-    if (!this.state.checklist.includes(relicId)) checklist.push(relicId);
+    if (!checklist.includes(relicId)) checklist.push(relicId);
     else checklist.splice(checklist.indexOf(relicId), 1);
     this.props.onChangeItems(checklist);
     localStorage.setItem("checklist", JSON.stringify(checklist));
     this.setState({checklist: checklist});
   }
 
+  clearChecklistAndLocalStorage() {
```
Comment (thisismydesign) [inline src/list.js]: I'd create separate methods or call this one something unified e.g. `clearState`.

File: src/list.js [hunk 2]
```diff
@@ -48,17 +48,25 @@ class List extends React.Component {
     };
 
     this.handleClick = this.handleClick.bind(this);
+    this.clearChecklistAndLocalStorage = this.clearChecklistAndLocalStorage.bind(this);
   }
 
   handleClick(relicId) {
     let checklist = this.state.checklist;
-    if (!this.state.checklist.includes(relicId)) checklist.push(relicId);
+    if (!checklist.includes(relicId)) checklist.push(relicId);
```
Comment (thisismydesign) [inline src/list.js:56]: Nice find!

**Comment by Kexort** [comment]
This PR has been merged as part of #44, right? @gomorizsolt @thisismydesign

**Comment by gomorizsolt** [comment]
Yes, I think so. #44 has the clear functionality as well I think, but I've renamed the function to "clearState" in this PR, that's why it has conflicts probably. Sorry, can't check, not at home yet.. :)

**Comment by Kexort** [comment]
I'll go ahead and close this then, we can reopen it if we missed something!

**Pull request closed without merging**